graph.py

```diff
@@ -81,6 +81,12 @@ def addUndirectedEdge(self, node:str, other:str, forwardCost:float=0, backwardCo
         # making sure a backward directed edge doesn't exist
         if self.edgeExists(other, node):
             raise Exception(f"There already exists a directed graph from {other} to {node}.")
+        
```

Comment:
We don't have to change the behavior of the graph Bini. More specifically, we want the graph to fail and complain if we tell it to make an edge between two nodes (so that we know we're doing sth wrong). Try not to touch the Graph class anywhere except the biDirectionalSearch method.

Reply:
the only thing I changed on the graph is on the method Graph.addUndirectedEdge()
if not self.nodeExists(node):
            self.addNode(node)
if not self.nodeExists(other):
            self.addNode(other)
I think this doesn't change the behavior of the graph, and I am avoiding repetitive work while adding the undirected edge.

Reply:
I think you are commenting on the first commit, incase you didn't see the second commit, please check it out.